<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Glossary</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>







<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Glossary</h1>



<div id="why-we-made-this-tutorial" class="section level2">
<h2>Why we made this tutorial</h2>
<p>This tutorial aims to clarify the nomenclature used in the urlb
package. There are a few methodological issues that are very specific to
microbial ecology and taxonomy; and that might be confusing for
non-microbiologists. On the other hand, there are a few terms from data
science and machine learning that may not be obvious for
microbiologists. And finally, of course, the nomenclature should be
understandable for researchers that focus on non-microbiome
datasets.</p>
</div>
<div id="phylogenetic-units" class="section level2">
<h2>Phylogenetic units</h2>
<p>We decided to use the term “phylogenetic units” instead of “species”.
Species would be the simplest term, but it is a problematic term in
microbial ecology. Briefly, microbial ecology requires the usage of
high-throughput sequencing based methods to distinguish microorganisms
in the environment; but usually it is hard to describe them down to
species level. Thus, we usually do not refer to “species” and instead
refer to the a proxy of species. This proxy can be, for example,
“amplicon sequence variants” (ASV), “operational taxonomic units” (OTU)
and so on. However, these terms are very specific to the methods used in
specific situations. To try and generalize, we compromised with the
“phylogenetic units”, because they represent the same idea independently
of the method used (and avoid the problem of the species
definition).</p>
<p>Another possibility would be “taxa/taxon”, which would be simple.
However, it could cause confusion, because of the different taxonomic
levels. Again, phylogenetic units might have their taxonomy more or less
resolved, but for the purposes of this package we do not care about
specific taxonomies, we care about what distinguishes different
entities, thus, the phylogenetic units.</p>
</div>
<div id="phylogenetic-units-translated-to-machine-learning" class="section level2">
<h2>Phylogenetic units translated to machine learning</h2>
<p>In terms of the machine learning algorithms (not just the ones used
in here), the phylogenetic units represent the observations (or
objects). Thus, each row should represent a phylogenetic unit, with a
column providing some ID for each phylogenetic unit.</p>
</div>
<div id="variables-and-features" class="section level2">
<h2>Variables and features</h2>
<p>For each phylogenetic unit we might have several variables, or
features, but the single feature that is considered for ulrb is the
abundance score. This abundance score will be different for each
methodology, study design, etc. However, there is always some sort of
abundance table with some sort of abundance score. For the purposes of
ulrb it does not matter how the score was obtained.</p>
<p>Extra variables are allowed, but they will be ignored for the purpose
of the actual algorithm. Except for the sample variable, which will be
used for grouping the phylogenetic units. Basically, the algorithm will
be applied for each sample provided independently. The
<strong>sample</strong> should not be confused with the
<strong>observation</strong> in this context. Think of the sample as a
physical representative of a community of phylogenetic units; and for
each phylogenetic unit you “observe” if it was present (with a given
abundance score) in said sample.</p>
<p>Additionally, the samples will provide extra variables, because they
might include environmental metadata. However, all of this additional
data will be ignored by the ulrb package. Thus, you can have an
abundance table in long format that includes all the available metadata
(can include environmental, methodological, taxa… details) and ulrb will
just add new columns with new variables – like the abundance
classification and details on the clustering result.</p>
</div>
<div id="abundance-classification" class="section level2">
<h2>Abundance classification</h2>
<p>The word “classification” can be ambiguous, because it might refer to
“taxonomic” classification, or the “abundance” classification. In ulrb,
we are dealing with a classification problem, because we do not know how
to <em>classify</em> phylogenetic units based on their abundance score.
It is after we use the classification from ulrb, which is the result of
a clustering, we can assert that the abundance classification of the
lowest abundance phylogenetic units represents the rare biosphere, and
so on. Thus, when we refer to “abundance classification” we are
referring to the actual result of the ulrb, by the function
<code>define_rb()</code>.</p>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<pre><code>#&gt; Warning in attr(x, &quot;align&quot;): &#39;xfun::attr()&#39; is deprecated.
#&gt; Use &#39;xfun::attr2()&#39; instead.
#&gt; See help(&quot;Deprecated&quot;)
#&gt; Warning in attr(x, &quot;format&quot;): &#39;xfun::attr()&#39; is deprecated.
#&gt; Use &#39;xfun::attr2()&#39; instead.
#&gt; See help(&quot;Deprecated&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Biology</th>
<th align="left">Microbiology</th>
<th align="left">Data.science</th>
<th align="left">ulrb</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Species</td>
<td align="left">OTU/ASV/etc</td>
<td align="left">Observation/object</td>
<td align="left">Phylogenetic unit</td>
</tr>
</tbody>
</table>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
